extends Area2D

var speed = 500
var direction = Vector2.ZERO

# PodmieÅ„ Å›cieÅ¼kÄ™ jeÅ›li inna
var SMALL_BULLET_SCENE = preload("res://Enemy/wybuchowy/bullet/maÅ‚y_bullet.tscn")

func _ready() -> void:
	# PoÅ‚Ä…cz sygnaÅ‚ z juÅ¼ istniejÄ…cego Timer node
	$Timer.connect("timeout", Callable(self, "_explode"))
	$Timer.start() # JeÅ›li autostart nie wÅ‚Ä…czony

func _physics_process(delta: float) -> void:
	$Sprite2D.rotation += 3 * delta  # wartoÅ›Ä‡ 3 = ok. 170 stopni/sek

	position += transform.x * speed * delta
	position += direction * speed * delta

func _on_body_entered(body: Node2D) -> void:
	if body.is_in_group("Player") and not Global.dash_immunity and not Global.immune:
		body.Dmg_Func(2)
		Global.immune = true
		queue_free()
	elif body.is_in_group("Wall"):
		_explode()
		queue_free()

func _explode() -> void:
	var directions = [
		Vector2.RIGHT,
		Vector2.LEFT,
		Vector2.UP,
		Vector2.DOWN,
		Vector2(1, 1),    
		Vector2(1, -1),   
		Vector2(-1, 1),   
		Vector2(-1, -1),  
	]

	for dir in directions:
		var small_bullet = SMALL_BULLET_SCENE.instantiate()
		

		get_parent().add_child(small_bullet)
		small_bullet.global_position = global_position
		# ðŸ§  Upewnij siÄ™, Å¼e ten maÅ‚y pocisk MA to pole `direction`
		small_bullet.direction = dir.normalized()

	queue_free()
